dex/elrond_dex_pair/src/lib.rs

```diff
@@ -429,10 +428,18 @@ pub trait Pair:
 
         reserve_token_in += &amount_in_after_fee;
         reserve_token_out -= &amount_out_optimal;
-        self.update_reserves(&reserve_token_in, &reserve_token_out, &token_in, &token_out);
+        self.set_virtual_reserves(
+            &token_in,
+            &token_in,
+            &token_out,
+            &reserve_token_in,
+            &reserve_token_out,
+        );
+        self.increase_token_reserve(&token_in, &amount_in_after_fee);
+        self.try_decrease_token_reserve(&token_out, &amount_out_optimal)?;
```

Comment:
Optional: This doesn't really matter, but it might be better to do the possibly error-returning call before the call that always works. (i.e. try decrease token_out, and then increase token_id)

Reply:
Done